### ISSUE TITLE
Multiplying copies of icons for symlinked directories with "Collapse directories" enabled

### ISSUE THREAD

**Issue by futurulus** [atom/tree-view#705]
1. Create a project directory with a symlink within itself:

    ```
    mkdir ~/some-test-dir/ && cd ~/some-test-dir/
    mkdir -p foo/bar
    ln -s foo/bar bar
    atom . &
    ```

2. Enable "Collapse directories" in tree-view settings.

    ![atom_step1](https://cloud.githubusercontent.com/assets/933453/12411826/893762b0-be36-11e5-9bb4-c38302beb9d7.png)

3. Right-click `some-test-dir` project folder > New file, save as `testfile.txt`. Result: `testfile.txt` is created (expected), another icon for `foo/bar` also shows up beneath it (bug).

    ![atom_step2](https://cloud.githubusercontent.com/assets/933453/12411828/893d60a2-be36-11e5-986b-88e469a489b5.png)

4. Modify testfile.txt and save. Result: two more icons for `foo/bar` appear.

    ![atom_step3](https://cloud.githubusercontent.com/assets/933453/12411827/893985b8-be36-11e5-80db-8acc6c880acb.png)

5. Repeat step 4 as desired until number of `foo/bar`'s is comically large. Two new icons appear each time the file is saved.

On Atom 1.2.3, tree-view 0.195.0, Ubuntu 14.04 LTS

**Comment by lostkhaos** [comment]
Can confirm, same exact issue. Running on Arch:

    Atom: 1.6.0-dev-1be4453
    tree-view: 0.200.0
    npm: 3.5.3
    node: 5.4.1

**Comment by CSomerville** [comment]
Partially reproduced this on OSX 10.10.2. OP's step 3 occurs as described, but not step 4. In any case, the symlink did not seem important, as step 3 occurred with or without it. Rather, the important thing is the name collision between the nested directory `foo/bar` and the other directory `bar`.

In `directory.coffee`, when the `squashDirectoryNames` method, responsible for handling the "Collapse Directories" feature, is called on `foo/bar`, it sets `@name` to `bar` and `@squashedName` to `foo/`:
https://github.com/atom/tree-view/blob/master/lib/directory.coffee#L276-L278

The `reload` method in the same file then uses `@name` to index child directories on the parent directory's `@entries` object. Because `@squashDirectoryNames` uses the nested folder as `@name`, there are two directory objects that can claim to be indexed by `@entries` under `bar`.

I uncovered a related issue, which is that, given a directory structure:
```
/some_test_dir/
  bar
  foo/baz
```
It seems that the `reload` method instantiates a new directory for `/foo/baz` and deletes the old one. The reason for this is in the `getEntries` method. This method uses `fs.readdir` to provide a list of `names` to check against `@entries`. However, this check fails:
https://github.com/atom/tree-view/blob/master/lib/directory.coffee#L178
because `fs.readdir` provides `foo` while the entry in question is indexed under `baz`. However, because of additional logic in the `reload` method, the old `baz` gets deleted and the new one takes its place.

Hope this helps. Best.

**Comment by RyanBalfanz** [comment]
I'm seeing this behavior with v1.8.0.

**Comment by nathansobo** [comment]
This will be fixed in 1.11.x thanks to @itiut.

**Comment by mdeboer** [comment]
Temporary fix until 1.11 is out is to disable "collapse directories".
